[PubSub] Clarify PubSub get started guide to have appropriate information to successfully work (#10624)
cloudflare/cloudflare-docs: content/pub-sub/guide.md

4. Choose **Get Started** next to **Create Custom Token** 
5. Name the token - e.g. "Pub/Sub Write Access"
6. Under the **Permissions** heading, choose **Account**, select **Pub/Sub** from the first drop-down, and **Edit** as the permission.

> 7. Select **Add More**  below the newly created permission. Choose **User**, select **Memberships** from the first drop-down, and **Edit** as the permission.

Reviewer: Can you explain why this is necessary / why it didn't work without this?

Author: Attempting to create a namespace with a token lacking these credentials fails with a 10000 authorization error for the API

This is something that I experience when working through the docs as is and has also been experienced by devs in discord doing this same

> 7. Select **Add More**  below the newly created permission. Choose **User**, select **Memberships** from the first drop-down, and **Edit** as the permission.

Reviewer 2: ```suggestion
7. Select **Add More** below the newly created permission. Choose **User** > **Memberships** from the first dropdown and **Edit** as the permission.
```

> 8. Click **Continue to Summary** at the bottom of the page, where you should see _All accounts - Pub/Sub:Edit_ as the permission

Reviewer 2: ```suggestion
8. Select **Continue to Summary** at the bottom of the page, where you should see _All accounts - Pub/Sub:Edit_ as the permission.
```

Author: @Maddy-Cloudflare why do we "click" some places and "select" others? What's the standard practice for this?
Note that "click" is the way the docs were previously written. I have no preference, just want to understand which goes where

Reviewer 2: Hi there, Charlie! Figured I'd respond while it's outside of working hours for Maddy. When the docs were originally written, the PCX team hadn't yet revisited guidance on Select vs Click. Select was the chosen option as it's device agnostic.

To avoid having that one step look completely different from the rest of the procedure, we can keep Click and just make note of revisiting the "Click" options to update them with Select at a later time.

Author: Thanks @dcpena makes perfect sense. I'll get a commit going changing that language for the next PR

> 9. Click **Create Token**, and copy the token value.

Reviewer 2: ```suggestion
9. Select **Create Token** and copy the token value.
```

$ npm i mqtt --save
```

Set your environment variables.

```sh

> #Tip: If you want these to persist across boots, store the below variables in your `.zshrc`, `.bashrc`, or `.profile` files in your home folder

Reviewer: ```suggestion
```

We generally want to avoid users from storing secrets persistently in config. Since this is a private beta, it's OK for these to be temporary.

$ export CLOUDFLARE_API_TOKEN="YourAPIToken"
$ export CLOUDFLARE_ACCOUNT_ID="YourAccountID"
$ export DEFAULT_NAMESPACE="TheNamespaceYouCreated"

> $ export BROKER_NAME="TheBrokerYouCreated"

Reviewer: It's likely better to have each of these done _before_ the `wrangler ...` command that uses that name, or immediately after, so they are in context.

Author: That's correct. 

The only time these env variables are used is in the command following this commit, with the exception of the CLOUDFLARE_API_TOKEN. Which is first exported above, but also is included below.

 I have repeated the API_TOKEN in this list for clarity, but am happy to remove that line if we feel it improves the reading experience

const brokerEndpoint = "mqtts://my-broker.my-namespace.cloudflarepubsub.com"
const options = {
  port: 8883,
  //If password alone does not successfully auth, uncomment the lines below and try again
  //clientID: "First portion of token generated above",

>   //username: "Your username",

Reviewer: I'd suggest just making this non-optional: don't make users decide during a tutorial.

Author: The reason these lines exist is specifically because of that `mqtt.js` silent auth failure bug we talked about yesterday. 

I added them in as comments to help anyone who might have trouble logging in, but if the bug is fixed then there's no need for them. Thought it would be a nice temporary measure. 

My preference would be removing these lines if we feel like that doesn't make sense.

Reviewer: I'd strongly recommend simplifying it. Reduce the choices a user has to make when starting out. "If that doesn't work, try again" is neither a) guaranteed to work and b) leads users to ask "why not just make that the default?"

Author: Good call. Will remove
